### PR TITLE
deposit and withdraw methods to interact with matic network

### PR DESCRIPTION
**Pull request by shahbhavir**
- Deposit
- Local withdraw
- Withdraw with backend APIs
- Tx/receipt/header proof and verify proof APIs
- Webpack setup
- Eslint fixes for future pull requests
- Necessary testcases
- Documentation

### CODE REVIEW

File: src/helpers/blocks.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+const sha3 = utils.sha3
+const BN = utils.BN
+
+export async function getHeaders(start, end, web3) {
+  if (start >= end) {
+    return []
+  }
+
+  let current = start
```
Comment (0xAshish) [inline src/helpers/blocks.js]: if we don't have any plan for getting current variable's value from somewhere else then using start variable seems ok.

Comment (jdkanani) [reply src/helpers/blocks.js]: @0xAshish context?

Comment (0xAshish) [reply src/helpers/blocks.js]: Minor thing : `let current = start` is not needed assignment, instead use `start` itself.

Comment (jdkanani) [reply src/helpers/blocks.js]: True

**Comment by jdkanani** [comment]
Related to #2

File: src/helpers/proofs.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  } catch (e) {
+    return false
+  }
+  return false
+}
+
+export function getReceiptBytes(receipt) {
+  let status = receipt.root
+  if (receipt.status !== undefined && receipt.status !== null) {
+    status = receipt.status
```
Comment (0xAshish) [inline src/helpers/proofs.js]: ` status = receipt.status ? "0x01" : "0x00" ` -- will work or why do we need to check for `else if` condition ?

File: src/helpers/proofs.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+      if (!nodeKey.equals(encodedNode)) {
+        return false
+      }
+      if (pathPtr > path.length) {
+        return false
+      }
+      switch (currentNode.length) {
+        case 17: // branch node
+          if (pathPtr === path.length) {
+            if (currentNode[16] === rlp.encode(value)) {
```
Comment (0xAshish) [inline src/helpers/proofs.js]: can we simply do  `return currentNode[16] === rlp.encode(value)` ?

File: src/helpers/proofs.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+          break
+        case 2:
+          pathPtr += nibblesToTraverse(
+            currentNode[0].toString("hex"),
+            path,
+            pathPtr
+          )
+          if (pathPtr === path.length) {
+            // leaf node
+            if (currentNode[1].equals(rlp.encode(value))) {
```
Comment (0xAshish) [inline src/helpers/proofs.js]: here as well `return currentNode[1].equals(rlp.encode(value)`
or if it's not boolean then `return !!currentNode[1].equals(rlp.encode(value)`

File: src/helpers/proofs.js (hunk 2 above)
Comment (0xAshish) [reply src/helpers/proofs.js]: @shahbhavir  would love your reviews here.

**Comment by jdkanani** [comment]
@ptsayli Please fix eslint

File: src/helpers/proofs.js (hunk 2 above)
Comment (shahbhavir) [reply src/helpers/proofs.js]: if `receipt.status` has been assigned a value, then the code pushed is correct, or else your suggestion would work too

**Comment by jdkanani** [comment]
@shahbhavir any luck?

**Pull request merged**